### PR TITLE
Refactor required parameters

### LINKED ISSUE #68: [spec compliance] Update parameters to reflect cnab-spec#227
(https://github.com/deislabs/cnab-spec/pull/227) in https://github.com/deislabs/cnab-spec introduced some changes to how parameters are defined, so we need to update cnab-go to reflect that.

### PR DESCRIPTION
**Pull request by jlegrone**
Relates to #68

### CODE REVIEW

File: bundle/parameters.go [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
-
-// ParameterDefinition defines a single parameter for a CNAB bundle
-type ParameterDefinition struct {
+// Parameter defines a single parameter for a CNAB bundle
+type Parameter struct {
 	Definition  string    `json:"definition" mapstructure:"definition"`
 	ApplyTo     []string  `json:"applyTo,omitempty" mapstructure:"applyTo,omitempty"`
 	Description string    `json:"description,omitempty" mapstructure:"description"`
 	Destination *Location `json:"destination,omitemtpty" mapstructure:"destination"`
+	Required    bool
```
Comment (vdice) [inline bundle/parameters.go]: Looks like this needs a corresponding json identifier still: `json:"required" mapstructure:"required"`

**Comment by jcsirot** [comment]
Since `Parameters` is now a map of `Parameter` I think that this check is not required anymore https://github.com/deislabs/cnab-go/blob/22ed003cf91e38449b08dfbf12ba1f5dc0f83344/bundle/bundle.go#L145

File: bundle/parameters_test.go [hunk 2]
```diff
@@ -39,16 +37,14 @@ func TestCanReadParameterDefinition(t *testing.T) {
...
-				"test": {
-					"definition": "%s",
-					"destination": {
-						"env": "%s",
-						"path": "%s"
-					},
-					"description": "%s",
-					"applyTo": [ "%s", "%s" ]
-				}
+			"test": {
```
Comment (radu-matei) [inline bundle/parameters_test.go]: Could you also add a small check for the required property here?

File: bundle/bundle.go [hunk 3]
```diff
@@ -97,7 +97,7 @@ type InvocationImage struct {
 	BaseImage `mapstructure:",squash"`
 }
 
-// Map that stores the relocated images
+// ImageRelocationMap stores the relocated images
```
Comment (radu-matei) [inline bundle/bundle.go]: Nice catch, thanks!

**Pull request merged**